test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -0,0 +1,268 @@
+    # already a very large test. This may be something to be added later when needed.
+    sdkmr = create_autospec(SDKMethodRunner, spec_set=True, instance=True)
+    logger = create_autospec(Logger, spec_set=True, instance=True)
+    ws = create_autospec(Workspace, spec_set=True, instance=True)
+    catutils = create_autospec(CatalogUtils, spec_set=True, instance=True)
+    catalog = create_autospec(Catalog, spec_set=True, instance=True)
+    condor = create_autospec(Condor, spec_set=True, instance=True)
+    kafka = create_autospec(KafkaClient, spec_set=True, instance=True)
+    mongo = create_autospec(MongoUtil, spec_set=True, instance=True)
+    slack = create_autospec(SlackClient, spec_set=True, instance=True)
+    sdkmr.get_logger.return_value = logger
+    sdkmr.get_workspace.return_value = ws
+    sdkmr.get_catalog_utils.return_value = catutils
+    catutils.get_catalog.return_value = catalog
+    sdkmr.get_condor.return_value = condor
+    sdkmr.get_kafka_client.return_value = kafka
+    sdkmr.get_slack_client.return_value = slack
+    sdkmr.get_mongo_util.return_value = mongo
+    sdkmr.get_user_id.return_value = "someuser"
+    sdkmr.get_token.return_value = "tokentokentoken"
+    sdkmr.check_as_admin.return_value = True
+    sdkmr.save_job.return_value = "603051cfaf2e3401b0500982"
+    ws.get_object_info3.return_value = {"paths": [["1/2/3"], ["4/5/6"]]}
+    catalog_resources = {
+        "client_group": "grotesquememlong",
+        "request_cpus": "4",
+        "request_memory": "32",
+    }
+    catutils.get_normalized_resources.return_value = catalog_resources
+    condor.extract_resources.return_value = CondorResources(
+        "4", "2600", "32", "grotesquememlong"
+    )
+    catalog.get_module_version.return_value = {"git_commit_hash": "git5678"}
+    condor.run_job.return_value = SubmissionInfo("cluster42", {}, None)
+    retjob = Job()
+    retjob.id = ObjectId("603051cfaf2e3401b0500982")
+    retjob.status = "created"
+    mongo.get_job.return_value = retjob
+
+    # set up the class to be tested and run the method
+    rj = EE2RunJob(sdkmr)
+    params = {
+        "method": "lolcats.lol_unto_death",
+        "source_ws_objects": ["1/2/3", "4/5/6"],
+    }
+    assert rj.run(params, as_admin=True) == "603051cfaf2e3401b0500982"
+
+    # check mocks called as expected
+    sdkmr.check_as_admin.assert_called_once_with(JobPermissions.WRITE)
+    ws.get_object_info3.assert_called_once_with(
+        {"objects": [{"ref": "1/2/3"}, {"ref": "4/5/6"}], "ignoreErrors": 1}
+    )
+    catutils.get_normalized_resources.assert_called_once_with("lolcats.lol_unto_death")
+    condor.extract_resources.assert_called_once_with(catalog_resources)
+    catalog.get_module_version.assert_called_once_with(
+        {"module_name": "lolcats", "version": "release"}
+    )
+
+    # initial job data save
+    expected_job = Job()
+    expected_job.user = "someuser"
+    expected_job.status = "created"
```

Comment:
is it not possible to do something like 
```py
expected_job = Job( "user": "some user", "status": "created")
```
or do all these objects have to be constructed very laboriously?

Reply:
I'm not super familiar with MongoEngine, but that's how it's done in their [tutorial](https://docs.mongoengine.org/tutorial.html#adding-data-to-our-tumblelog) as well as in the EE2 code, so my assumption is not.

Reply:
Yes, in that link you posted you can see that you can do something like
```
>>> class User(Document):
...     email = StringField(required=True)
...     first_name = StringField(max_length=50)
...     last_name = StringField(max_length=50)
...
>>> u = User(email='example@example.com', first_name='bob', last_name='dole')
>>> u
<User: User object>
```

Reply:
Ha, I guess my eyes were drawn to the big old note and that confirmed what I suspected so I moved on